tests/subsys/net/lib/nrf_cloud/cloud/src/main.c

```diff
@@ -0,0 +1,1749 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include "fakes.h"
+#if defined(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD)
+#include "nrf_cloud.c"
+#endif
+
+/* Flags to indicate if the cloud state */
+K_SEM_DEFINE(cloud_connected, 0, 1);
+K_SEM_DEFINE(cloud_disconnected, 0, 1);
+K_SEM_DEFINE(dc_connected, 0, 1);
+K_SEM_DEFINE(cc_connected, 0, 1);
+static atomic_t transport_connecting;
+uint32_t status;
+
+/* Simple event handler needed for the init function to
+ * have a successful initialization in the test below
+ */
+void event_handler(const struct nrf_cloud_evt *const evt)
+{
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_READY:
+		k_sem_give(&dc_connected);
+		break;
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECTED:
+		k_sem_reset(&cloud_disconnected);
+		k_sem_give(&cloud_connected);
+		break;
+	case NRF_CLOUD_EVT_RX_DATA_GENERAL:
+		k_sem_give(&cc_connected);
+		break;
+	case NRF_CLOUD_EVT_TRANSPORT_DISCONNECTED:
+		if (evt->status != 0) {
+			atomic_set(&transport_connecting, 0);
+			status = evt->status;
+		} else {
+			k_sem_reset(&dc_connected);
+			k_sem_reset(&cc_connected);
+			k_sem_reset(&cloud_connected);
+			k_sem_give(&cloud_disconnected);
+		}
+		break;
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECTING:
+		atomic_set(&transport_connecting, 1);
+		status = evt->status;
+		break;
+	case NRF_CLOUD_EVT_ERROR:
+		printk("Cloud error: %d\n", evt->status);
+		break;
+	default:
+		break;
+	}
+}
+
+#if defined(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD)
+/* Wait for the connecting event for 2s, timeout after that.
+ * Then wait for disconnected event for 2s, timeout after that.
+ * Use with testing nrf_cloud_run for POLL THREAD only
+ */
+void wait_for_connecting_and_disconnected_poll_thread(void)
+{
+	uint32_t wait_counter = 10;
+
+	while (--wait_counter && !atomic_get(&transport_connecting)) {
+		k_sleep(K_MSEC(200));
+	}
+	zassert_not_equal(0, wait_counter, "cloud connecting timed out");
+
+	/* Wait for the disconnected event for 2s, timeout after that */
+	wait_counter = 10;
+	while (--wait_counter && atomic_get(&transport_connecting)) {
+		k_sleep(K_MSEC(200));
+	}
+	zassert_not_equal(0, wait_counter, "cloud disconnect timed out");
+}
+#endif
+
+/* Global param to use in nrf_cloud_init */
+const struct nrf_cloud_init_param init_param = {
+	.event_handler = event_handler,
+	.client_id = NULL,
+};
+
+/* Global sensor data for sending to cloud */
+const struct nrf_cloud_data cloud_data = {
+	.len = 0U,
+	.ptr = NULL,
+};
+
+/* Global sensor param to use in
```

Comment:
note: some of your comment lines are very short... you are allowed up to 100 characters per line.

Reply:
Haha, I like to make the lines of the comment similar in length, just a habit.

Reply:
changed them